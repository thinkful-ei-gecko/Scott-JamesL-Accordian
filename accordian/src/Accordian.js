import React from 'react';

export default class Accordian extends React.Component{
  static defaultProps = {sections: []};
  state = {
    currentIndex: null
  };

  handleButtonClick = index => {
    if(index === this.state.currentIndex){
      this.setState({
        currentIndex: null
      })
    }else{
      this.setState({
        currentIndex: index
      })
    }
  }

  renderButtons() {
    return this.props.sections.map((section, index) => (
      <li key = {index}>
        <button onClick={() => this.handleButtonClick(index)}>
          {section.title}
        </button>
        <p>{this.state.currentIndex === index ? section.content : ''}</p>
      </li>
    ))
  }

  generateLi(){
    //return arr of items to gen li elems

  }


  render() {

    return (
      <ul>
        {this.renderButtons()}
      </ul>
    )
  }


}