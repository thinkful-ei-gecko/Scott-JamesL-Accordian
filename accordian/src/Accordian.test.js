import React from 'react';
import ReactDOM from 'react-dom';
import Accordian from './Accordian';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe(`<Accordian />`, () => {
    it('renders without errors when props is empty', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty ul', () => {
        const tree = renderer.create(<Accordian />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders with no sections active', () => {
        const tree = renderer.create(<Accordian sections={sectionsData}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('renders empty given no tabs', () => {
        const wrapper = shallow(<Accordian />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })

    it('renders with all sections closed by default', () => {
        const wrapper = shallow(<Accordian sections={sectionsData}/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('expands section on click', () => {
        const wrapper = shallow(<Accordian sections={sectionsData} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('only has one section open after multiple clicks', () => {
        const wrapper = shallow(<Accordian sections={sectionsData} />)
        wrapper.find('button').at(2).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(0).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})

const sectionsData = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]