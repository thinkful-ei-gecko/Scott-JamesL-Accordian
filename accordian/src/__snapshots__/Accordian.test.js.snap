// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordian /> expands section on click 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`<Accordian /> only has one section open after multiple clicks 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`<Accordian /> renders empty given no tabs 1`] = `<ul />`;

exports[`<Accordian /> renders empty ul 1`] = `<ul />`;

exports[`<Accordian /> renders with all sections closed by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;

exports[`<Accordian /> renders with no sections active 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      
    </p>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      
    </p>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      
    </p>
  </li>
</ul>
`;
